keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -53,22 +66,22 @@ class ArtifactListener(
       artifactRepository.register(artifact)
     }
 
-    if (artifactRepository.versions(artifact).isEmpty()) {
-      storeLatestVersion(artifact, event.statuses)
+    if (artifact.type == DEB && artifactRepository.versions(artifact).isEmpty()) {
```

Comment:
nit: Any reason to use one check for type (`korkArtifact.isDeb()`, etc.) above and a different check here? I prefer the check style here. We could replace the checks above with this style as well for consistency.

Reply:
They're on two different objects - a `[korkArtifact](https://github.com/spinnaker/kork/blob/master/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/model/Artifact.java)` is the generic artifact payload, where everything important is a string. This artifact is a `DeliveryArtifact` (or something like that) where the type field is an enum. So, we need two different checks unfortunately.

Reply:
I know, but on line 33 you already have the `DeliveryArtifact` object and so all the checks inside the `when` block could be using it, which would make for the same type of check as you have here...

Reply:
Ah good call! I'll just use that 🤦‍♂